smdebug/exceptions.py

```diff
@@ -1,58 +1,71 @@
 # First Party
+from smdebug.core.logger import get_logger
 from smdebug.core.modes import ModeKeys as modes
 
 
-class InvalidCollectionConfiguration(Exception):
+class SMDebugError(Exception):
+    def __init__(self):
+        logger = get_logger("exceptions")
+        logger.error("%s: %s", self.__class__.__name__, str(self))
+        for h in logger.handlers:
+            h.flush()
+
+
+class InvalidCollectionConfiguration(SMDebugError):
     def __init__(self, c_name):
         self.c_name = c_name
+        super().__init__()
 
     def __str__(self):
-        return f"Collection {self.c_name} has not been configured. \
-        Please fill in tensor_name or include_regex"
+        return f"Collection {self.c_name} has not been configured. Please fill in tensor_name or include_regex"
 
 
-class StepNotYetAvailable(Exception):
+class StepNotYetAvailable(SMDebugError):
```

Comment:
Depending on what we want to achieve from /opt/../failure, we need to review this, what is treated as error in logs and what is info.